### PR TITLE
Make terraform module variables nullable

### PR DESCRIPTION
**Pull request by johnake**
## Context
So we can have the option of leaving some parameters empty and they pick the default value or we override the default other values when we use the modules

## Changes proposed in this pull request
Make terraform module variables nullable if they are optional and have default values(and the default value isn't null)

## Checklist

- [x] I have performed a self-review of my code, including formatting and typos
- [x] I have [cleaned the commit history](https://www.annashipman.co.uk/jfdi/good-pull-requests.html)
- [x] I have added the `Devops` label
- [x] I have attached the pull request to the trello card

### CODE REVIEW

**Comment by johnake** [comment]
[Make terraform module variable nullable](https://trello.com/c/6553GFop/808-make-terraform-module-variable-nullable)

File: aks/application/variables.tf [hunk 1]
```diff
@@ -82,24 +85,28 @@ variable "is_web" {
 
 variable "web_external_hostnames" {
   type        = list(string)
+  nullable    = true
```
Comment (thomasleese) [inline aks/application/variables.tf:88]: I wonder if lists should really be `nullable`, what's the difference between the default `[]` and passing in an explicit `null`?

File: aks/redis/variables.tf [hunk 2]
```diff
@@ -71,13 +72,15 @@ variable "azure_sku_name" {
...
+  type     = string
+  nullable = true
+  default  = "1.2"
 }
 
 variable "azure_public_network_access_enabled" {
-  type    = bool
-  default = false
+  type     = bool
+  nullable = true
```
Comment (thomasleese) [inline aks/redis/variables.tf:82]: I don't think booleans should be `nullable` since `false` acts the same as `null`.

**Review by thomasleese — COMMENTED** [review]
Should we instead be setting `nullable = false` on the required variables? I believe [`nullable = true`](https://developer.hashicorp.com/terraform/language/values/variables#disallowing-null-input-values) is the default for all variables.

File: aks/redis/variables.tf (hunk 2 above)
Comment (johnake) [reply aks/redis/variables.tf]: I had updated to nullable = false. 
See this https://github.com/hashicorp/terraform/pull/29832#issue-1039803860, for reference

**Pull request merged**